lib/remoteproc/remoteproc.c

```diff
@@ -29,10 +29,11 @@ remoteproc_check_fw_format(const void *img_data, size_t img_len)
 		return NULL;
 }
 
+/* try the internal list added by remoteproc_add_mem first and then get_mem callback */
 static struct remoteproc_mem *
 remoteproc_get_mem(struct remoteproc *rproc, const char *name,
 		   metal_phys_addr_t pa, metal_phys_addr_t da,
-		   void *va, size_t size)
+		   void *va, size_t size, struct remoteproc_mem *buf)
```

Comment:
'buf' is unused in this function, except in the (so far non-existent) case where the get_mem operation is provided. This may cause warnings from static analyzers. Maybe it would be better to always set it to match the return value.

Reply:
> 'buf' is unused in this function, except in the (so far non-existent) case where the get_mem operation is provided. This may cause warnings from static analyzers.

Could you point me the warning output? 

> Maybe it would be better to always set it to match the return value.

But it require we copy the whole struct from 'mem' to 'buf', which is unnecessary.

Reply:
Anyway, I add the follow code like others:
```
(void)buf;
```
which should address your concern.